src/test/java/bio/terra/service/auth/iam/ras/EcmServiceTest.java

```diff
@@ -0,0 +1,152 @@
+  }
+
+  @Test
+  public void testGetRasDbgapPermissions() throws Exception {
+    // Note: many of the "unhappy paths" in this test should not occur if ECM only returns valid
+    // passports and visas as indicated by their Swagger documentation.
+    // Testing them anyway to verify our own behavior.
+    doAnswer(a -> null).when(ecmService).getRasProviderPassport(authenticatedUserRequest);
+    assertThat(
+        "No RAS dbGaP permissions when a user doesn't have a passport",
+        ecmService.getRasDbgapPermissions(authenticatedUserRequest),
+        empty());
+
+    doAnswer(a -> toJwtToken("{}"))
+        .when(ecmService)
+        .getRasProviderPassport(authenticatedUserRequest);
+    assertThat(
+        "No RAS dbGaP permissions when a user's passport has no visa claim'",
+        ecmService.getRasDbgapPermissions(authenticatedUserRequest),
+        empty());
+
+    String passportPayloadNoVisas =
+        String.format("{\"%s\":[]}", EcmService.GA4GH_PASSPORT_V1_CLAIM);
+    doAnswer(a -> toJwtToken(passportPayloadNoVisas))
+        .when(ecmService)
+        .getRasProviderPassport(authenticatedUserRequest);
+    assertThat(
+        "No RAS dbGaP permissions when a user's passport has no visas",
+        ecmService.getRasDbgapPermissions(authenticatedUserRequest),
+        empty());
+
+    String invalidVisaJwt = toJwtToken(loadJson("ga4gh_passport_v1_invalid.json"));
+    String invalidPassportPayload =
+        String.format("{\"%s\":[\"%s\"]}", EcmService.GA4GH_PASSPORT_V1_CLAIM, invalidVisaJwt);
+    String invalidPassportJwt = toJwtToken(invalidPassportPayload);
+    doAnswer(a -> invalidPassportJwt)
+        .when(ecmService)
+        .getRasProviderPassport(authenticatedUserRequest);
+    assertThat(
+        "No RAS dbGaP permissions when a user's passport has invalid visas",
+        ecmService.getRasDbgapPermissions(authenticatedUserRequest),
+        empty());
+
+    String visaJwt = toJwtToken(loadJson("ga4gh_passport_v1_valid.json"));
+    String passportPayload =
+        String.format("{\"%s\":[\"%s\"]}", EcmService.GA4GH_PASSPORT_V1_CLAIM, visaJwt);
+    String passportJwt = toJwtToken(passportPayload);
+    doAnswer(a -> passportJwt).when(ecmService).getRasProviderPassport(authenticatedUserRequest);
+    assertThat(
+        "Passport visa permissions are successfully decoded and unknown properties ignored",
+        ecmService.getRasDbgapPermissions(authenticatedUserRequest),
+        Matchers.contains(
+            new RasDbgapPermissions("c01", "phs000001"),
+            new RasDbgapPermissions("c02", "phs000001"),
+            new RasDbgapPermissions("c02", "phs000002")));
+  }
+
+  /** Helper Methods for Generating JWT Tokens * */
+  private String loadJson(String filename) throws Exception {
+    return jsonLoader.loadJson(String.format("./service/auth/iam/ras/%s", filename));
+  }
+
+  private String toJwtToken(String payload) throws Exception {
+    String header = "{\"alg\":\"none\",\"typ\":\"JWT\"}";
```

Comment:
You could use a text block here and elsewhere to avoid the quoting
```suggestion
    String header = """
            {"alg":"none","typ":"JWT"}""";
```